actors/evm/src/interpreter/instructions/call.rs

```diff
@@ -157,50 +164,77 @@ pub fn call<BS: Blockstore, RT: Runtime<BS>>(
-                    TokenAmount::from(&value),
-                ),
-
-                CallKind::DelegateCall => {
-                    // first invoke GetBytecode to get the code CID from the target
-                    let code = crate::interpreter::instructions::ext::get_evm_bytecode_cid(
-                        system.rt, dst,
-                    )?;
-
-                    // and then invoke self with delegate; readonly context is sticky
-                    let params = DelegateCallParams {
-                        code,
-                        input: input_data.to_vec(),
-                        readonly: system.readonly,
-                    };
-                    system.send(
-                        &system.rt.message().receiver(),
-                        Method::InvokeContractDelegate as u64,
-                        RawBytes::serialize(&params)?,
-                        TokenAmount::from(&value),
-                    )
+            // Special casing for embryo/non-existent actors: we just do a SEND (method 0)
+            // which allows us to transfer funds (and create embryos)
+            let is_embryonic = if dst_addr.protocol() == AddressProtocol::ID {
+                // sanity check: this shouldn't be an ID address, as you can't predict
+                // what actor is gonna sit there.
+                false
+            } else if let Some(actor_id) = system.rt.resolve_address(&dst_addr) {
+                if let Some(cid) = system.rt.get_actor_code_cid(&actor_id) {
+                    system.rt.resolve_builtin_actor_type(&cid) == Some(Type::Embryo)
+                } else {
+                    true
                 }
+            } else {
+                true
+            };
 
-                CallKind::StaticCall => system.send(
+            let call_result = if is_embryonic {
+                system.send(
                     &dst_addr,
-                    Method::InvokeContractReadOnly as u64,
-                    // TODO: support IPLD codecs #758
+                    0,
+                    // we still send the input, even thought it will be ignored, for debugging
+                    // purposes
                     RawBytes::serialize(BytesSer(input_data))?,
                     TokenAmount::from(&value),
-                ),
+                )
+            } else {
+                match kind {
+                    CallKind::Call => system.send(
+                        &dst_addr,
+                        // readonly is sticky
+                        if system.readonly {
+                            Method::InvokeContractReadOnly
+                        } else {
+                            Method::InvokeContract
+                        } as u64,
+                        // TODO: support IPLD codecs #758
+                        RawBytes::serialize(BytesSer(input_data))?,
+                        TokenAmount::from(&value),
+                    ),
+
+                    CallKind::DelegateCall => {
```

Comment:
Not convinced we've covered the case of DELEGATECALL to an EOA correctly. I think it should be handled on the embryonic block, and possibly should noop (as the callee would execute in our context), but this needs deeper analysis. It's quite an edge case, so I'm happy merging as is. I'll open a TODO issue in ref-fvm.

Reply:
Ah, nevermind. `DELEGATECALL` does _not_ take a value, so we're good here!